ui/shared/components/panel/variants/VariantClassify.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Button, Label } from 'semantic-ui-react'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+import { HttpRequestHelper } from 'shared/utils/httpRequestHelper'
+import { ColoredIcon } from 'shared/components/StyledComponents'
+import AcmgModal from '../acmg/AcmgModal'
+import { VerticalSpacer } from '../../Spacers'
+import PopupWithModal from '../../PopupWithModal'
+import { getVariantMainTranscript } from '../../../utils/constants'
+
+const CLINGEN_ALLELE_REGISTRY_URL = 'http://reg.genome.network/allele'
+const CLINGEN_VCI_URL = 'https://curation.clinicalgenome.org/select-variant'
+
+const ClinGenAlleleId = ({ alleleId, copied, onCopy }) => (
+  <CopyToClipboard
+    text={alleleId}
+    onCopy={onCopy}
+  >
+    <div>
+      <span>
+        {alleleId}
+        &nbsp;
+      </span>
+      {copied ? <ColoredIcon name="check circle" color="#00C000" /> : <ColoredIcon name="copy" link />}
+    </div>
+  </CopyToClipboard>
+)
+
+ClinGenAlleleId.propTypes = {
+  alleleId: PropTypes.string,
+  copied: PropTypes.bool,
+  onCopy: PropTypes.func,
+}
+
+const getButtonBackgroundColor = (classification) => {
+  const categoryColors = {
+    Unknown: 'grey',
+    Benign: 'blue',
+    'Likely Benign': 'blue',
+    Pathogenic: 'orange',
+    'Likely Pathogenic': 'orange',
+    Uncertain: 'yellow',
+  }
+  return categoryColors[classification] || 'grey'
+}
+
+class VariantClassify extends React.PureComponent {
+
+  static propTypes = {
+    variant: PropTypes.object.isRequired,
+    familyGuid: PropTypes.string,
+  }
+
+  state = {
+    copied: false,
+    alleleId: null,
+  };
```

Comment:
Change from a function to the PureComponent class to store states. The content of the `Popup` component re-renders every time it is displayed. The states are lifted from the popup content components to maintain the states between two popup events.

Reply:
This is unnecessarily complex and a premature optimization. It is better to refetch the Allele Id every time the Popup opens. The component that handles fetching and displaying the allele ID should be rendered inside the popup content. To prevent accidentally opening the popup, we should make this popup only triggered on click instead of hover which means its unlikely it will ever be opened except the one time someone wants to use it

Reply:
Okay. I will change it to be triggered on click.